### PR TITLE
Fix rancher-version and kube-version annotations

### PR DESCRIPTION
**Pull request by PennyScissors**
This PR cherry picks and squashes the commits from the PRs below (all reviewed and approved by their respective teams) to consolidate them and make the release easier:
team1 https://github.com/rancher/charts/pull/1926
team2 https://github.com/rancher/charts/pull/1927
team3 https://github.com/rancher/charts/pull/1928
team4 https://github.com/rancher/charts/pull/1923
infracloud https://github.com/rancher/charts/pull/1924
longhorn https://github.com/rancher/charts/pull/1929
harvester https://github.com/rancher/charts/pull/1925

### CODE REVIEW

File: charts/harvester-csi-driver/100.0.0+up0.1.9/Chart.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 annotations:
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Harvester CSI Driver
-  catalog.cattle.io/kube-version: '>= 1.18'
+  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.25.0-0'
```
Comment (aiyengar2) [inline charts/harvester-csi-driver/100.0.0+up0.1.9/Chart.yaml:4]: same comment as above here, not sure if this initial chart was tested on 1.24 https://github.com/rancher/charts/pull/1925#discussion_r903262223

**Review by aiyengar2 — APPROVED** [review]
PHEW! that's a huge PR. Approving but have two last minute questions about harvester stuff; if we can confirm with their team before merging, this LGTM

File: charts/harvester-cloud-provider/100.0.0+up0.1.8/Chart.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 annotations:
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Harvester Cloud Provider
-  catalog.cattle.io/kube-version: '>= 1.18'
+  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.25.0-0'
```
[the remark this answers is not in the document]
Comment (PennyScissors) [reply charts/harvester-cloud-provider/100.0.0+up0.1.8/Chart.yaml:4]: updated

File: charts/harvester-csi-driver/100.0.0+up0.1.9/Chart.yaml (hunk 1 above)
Comment (PennyScissors) [reply charts/harvester-csi-driver/100.0.0+up0.1.9/Chart.yaml:4]: updated

**Pull request merged**